crates/edr_provider/tests/integration/eip7623/send_data_to_eoa.rs

```diff
@@ -0,0 +1,54 @@
+use edr_eth::{address, bytes, transaction::EthTransactionRequest, SpecId};
+use edr_rpc_eth::CallRequest;
+
+use super::new_provider;
+use crate::integration::eip7623::assert_transaction_gas_usage;
+
+fn call_request() -> CallRequest {
+    let transaction_request = transaction_request();
+
+    CallRequest {
+        from: Some(transaction_request.from),
+        to: transaction_request.to,
+        data: transaction_request.data,
+        ..CallRequest::default()
+    }
+}
+
+fn transaction_request() -> EthTransactionRequest {
+    EthTransactionRequest {
+        from: address!("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"),
+        to: Some(address!("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266")),
+        data: Some(bytes!("0x11")),
+        ..EthTransactionRequest::default()
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn estimate_gas() -> anyhow::Result<()> {
+    let cancun_provider = new_provider(SpecId::CANCUN)?;
+    assert_eq!(
+        super::estimate_gas(&cancun_provider, call_request()),
+        // NOTE: Our estimate differs from the real cost by 1 gas unit.
+        21_017
```

Comment:
@fvictorio our gas estimate is off by one. Is that expected?

Reply:
Could this be caused by https://github.com/NomicFoundation/edr/blob/404821a26234bf67c4772add83c7e593c086ec42/crates/edr_provider/src/data.rs#L788-L790 ?

Reply:
Tbh I don't remember why we added that in the first place, so I can't argue if it's desirable or not. But removing it would be risky unless we are sure it won't have an impact. The semantics of that method are "if you use this gas limit, then your transaction won't OOG, and whatever gas ends up being used is not going to be significantly lower than the one yo uset", so adding one wei is unlikely to be a problem.

Reply:
I'll leave it as is then